packages/lesswrong/server/repos/UsersRepo.ts

```diff
@@ -346,12 +362,41 @@ export default class UsersRepo extends AbstractRepo<DbUser> {
           SELECT val
           FROM UNNEST(ARRAY_AGG(agreement_value)) AS val
           WHERE val != 0
-      ) AS agreement_values
-    FROM "CombinedVotes"
-    GROUP BY user_id, user_username, user_displayName
+      ) AS agreement_values,
+      uc.recently_active_matchmaking
+    FROM "CombinedVotes" as cv
+    LEFT JOIN "UserChecks" AS uc ON cv.user_id = uc._id
+    GROUP BY 
+      user_id, 
+      user_username, 
+      user_displayName,
+      uc.recently_active_matchmaking
     HAVING SUM(vote_power) > 1
     ORDER BY total_power DESC
     LIMIT $4;
-      `, [user._id, smallVotePower, bigVotePower, limit])
+      `, [user._id, smallVotePower, bigVotePower, limit, recencyLimitDays])
+  }
+
+
+  async getDialogueMatchedUsers(userId: string): Promise<DbUser[]> {
+    return this.any(`
+      SELECT DISTINCT(u.*)
+      FROM "DialogueChecks" other_users_checks
+      JOIN "DialogueChecks" current_user_checks
+      -- Join such that there must exist reciprocal checks
+      ON (
+        other_users_checks."targetUserId" = current_user_checks."userId"
+        AND current_user_checks."targetUserId" = other_users_checks."userId"
+        AND other_users_checks.checked IS TRUE
```

Comment:
I am of the opinion these IS TRUE clauses should be in a WHERE clause,
Same for 398 below. 397 is redundant.

Reply:
Leaving for now to ship sooner